### ISSUE TITLE
Index and count must refer to a location within the string

### ISSUE THREAD

**Issue by isaacabraham** [rogeralsing/Wire#109]
Apologies for the lack of a repro here. On a customer site, trying Wire out for an array of C# POCOs with some nullable fields, and properties that use object hierarchies, getting the following error:

`Index and count must refer to a location within the string: Parameter name: s`

Afraid I can't give you the actual type structure or sample data - I've tried reproducing myself but couldn't so far. Have you seen this before?

```
   at System.Text.UTF8Encoding.GetBytes(String s, Int32 charIndex, Int32 charCount, Byte[] bytes, Int32 byteIndex)
   at Wire.NoAllocBitConverter.GetBytes(String str, SerializerSession session, Int32& byteCount)
   at Wire.ValueSerializers.StringSerializer.WriteValueImpl(Stream stream, String s, SerializerSession session)
   at Wire.ValueSerializers.StringSerializer.WriteValue(Stream stream, Object value, SerializerSession session)
   at lambda_method(Closure , Stream , Object , SerializerSession )
   at Wire.ValueSerializers.ObjectSerializer.WriteValue(Stream stream, Object value, SerializerSession session)
   at Wire.Extensions.StreamEx.WriteObject(Stream stream, Object value, Type valueType, ValueSerializer valueSerializer, Boolean preserveObjectReferences, SerializerSession session)
   at lambda_method(Closure , Stream , Object , SerializerSession )
   at Wire.ValueSerializers.ObjectSerializer.WriteValue(Stream stream, Object value, SerializerSession session)
   at Wire.Extensions.StreamEx.WriteObject(Stream stream, Object value, Type valueType, ValueSerializer valueSerializer, Boolean preserveObjectReferences, SerializerSession session)
   at Wire.SerializerFactories.ArraySerializerFactory.WriteValues[T](T[] array, Stream stream, Type elementType, ValueSerializer elementSerializer, SerializerSession session)
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid6[T0,T1,T2,T3,T4,T5](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5)
   at Wire.SerializerFactories.ArraySerializerFactory.<>c__DisplayClass4_0.<BuildSerializer>b__1(Stream stream, Object arr, SerializerSession session)
   at Wire.ValueSerializers.ObjectSerializer.WriteValue(Stream stream, Object value, SerializerSession session)
   at Wire.Serializer.Serialize(Object obj, Stream stream)
```

**Comment by isaacabraham** [comment]
Note - the same object graph works with `BinaryFormatter`.

**Comment by rogeralsing** [comment]
Not seen that one before.
Any chance you could get hold of some sample data for this?

**Comment by rogeralsing** [comment]
I think I've found it #110 
I was incorrectly assuming that the byte count should be used, but it was char count.
I have however no data that triggers the problem so I can't verify yet.

**Comment by isaacabraham** [comment]
I'll build locally and test with that and get back to you. Cheers.

**Comment by isaacabraham** [comment]
@rogeralsing the solution doesn't build for me locally - I've just tried opening it and building, but it complains about not being able to download a load of NuGet packages e.g. System.Security.Cryptography.Algorithms. I suspect that this is because of framework incompatibilities e.g. NET45 etc..

I converted over to Paket, which did then manage to download the dependencies, but I was then left with another build error regarding ambigious calls to multiple methods with the same signature (e.g. ReflectionEx methods).
